### ISSUE TITLE
bun build fails: The superclass is not a constructor

### ISSUE THREAD

**Issue by laurensV** [oven-sh/bun#8925]
### What version of Bun is running?

1.0.26+c75e768a6

### What platform is your computer?

Linux 5.15.133.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

Some packages run just fine when using bun directly, but when trying to use `bun build` it gives this error `TypeError: The superclass is not a constructor.`

I created a small [example repo](https://github.com/laurensV/bun-build-bug-solana-web3js):
First install the package, for example `@solana/web3.js`
```
bun install @solana/web3.js
```
create file `index.js` that uses the package:
```
import * as web3 from '@solana/web3.js';
console.log(web3);
```

Running this file directly with `bun index.js` works fine, but when trying to build first and then run it (or when using `--compile` flag to put it into one executable), it fails:
```
bun build ./index.js --outdir ./build --target=bun && bun build/index.js
```
```
13821 | var fetchImpl = typeof globalThis.fetch === "function" ? globalThis.fetch : async function(input, init) {
13822 |   const processedInput = typeof input === "string" && input.slice(0, 2) === "//" ? "https:" + input : input;
13823 |   return await nodeFetch.default(processedInput, init);
13824 | };
13825 | 
13826 | class RpcWebSocketClient extends client.default {
        ^
TypeError: The superclass is not a constructor.
      at /$bunfs/root/test:13826:1
```
Seen this behaviour in multiple packages now, like puppeteer (https://github.com/oven-sh/bun/issues/6261) and AWS CDK (https://github.com/oven-sh/bun/issues/3088)

You can find the example repo code here: https://github.com/laurensV/bun-build-bug-solana-web3js

**Comment by tim-brand** [comment]
I got the same issue, with the `puppeteer` package.

**Comment by Cyenoch** [comment]
This problem was solved when I installed the dependencies through `npm install` instead of `bun i`. I thought there might be some unexpected problems with `bun i`.

**Comment by toondkn** [comment]
Observing the same incorrect behavior when trying to build code importing the `jspdf` package.

**Comment by toondkn** [comment]
Upon further testing, at least for the `jspdf` package, it appears the issue arises when build with `bun build [...] --target=bun`. When using `bun build [...] --target=node`, it works fine, even when the bundled code is executed with bun.

**Comment by toondkn** [comment]
After more testing, the bundle built with `--target=node` starts without errors, but it still errors at runtime when constructing the same class that causes parsing issues on startup when bundled with `--target=bun`.

**Comment by liquidg3** [comment]
Same problem without our CLI tools here: https://github.com/sprucelabsai-community/spruce-cli-workspace

1. `cd packages/spruce-cli`
2. `bun build ./src/index.ts --compile --outfile=spruce`
3. `./spruce`

**Comment by avarayr** [comment]
Experiencing same issue
```
bun build ./index.js --compile --outfile mycli
```
```bash
❯ ./mycli
45513 |     this._process();
45514 |     return this._hash;
45515 |   }
45516 | }
45517 |
45518 | class _Readable extends EventEmitter2 {
        ^
TypeError: The superclass is not a constructor.
      at /$bunfs/root/mycli:45518:1
```

**Comment by nephix** [comment]
I got the same issue in when using `jsx-email` (check this [issue](https://github.com/shellscape/jsx-email/issues/197) with minimal reproduction example: https://github.com/shellscape/jsx-email/issues/197)

It works fine with
```
bun run index.ts
```

but fails with

```
bun run build
bun dist/index.js
```

**Comment by risen228** [comment]
Encountered the same issue with `W3CBaggagePropagator` when using `@sentry/bun`

Bun version: v1.1.20

```js
await Bun.build({
  entrypoints: ['src/main.ts'],
  outdir: 'dist',
  target: 'bun',
  sourcemap: 'external',
})
```

**Comment by kaitakami** [comment]
Same issue with `llamaindex` when using `OpenAI`.

import
```ts
import { OpenAI } from "llamaindex";
```

```ts
const openaiLLM = new OpenAI({ model: "gpt-4o-mini", temperature: 0.7 });
```

**Comment by Jarred-Sumner** [comment]
This was fixed in Bun v1.1.21. To get Bun v1.1.21, please run

```
bun upgrade
```

If you're still running into this issue, please leave a comment , we'll re-open it and @paperdave will investigate

**Comment by michaelessiet** [comment]
Still running into this issue with `@solana/web3.js` after building and running the code using bun
